mevislab.github.io/content/tutorials/image_processing/image_processing6.md

```diff
@@ -4,194 +4,174 @@ date: 2025-05-05
+There are no names for the contours by default, so we need the `CSOLabelRenderer` module to add labels (e.g., 'Bladder', 'Prostate') next to each contour, helping to clearly identify the anatomy. The figure below shows that :
 
-![panel CSOLabelRenderer to name the contour](/images/tutorials/image_processing/panelCSOLabelRenderertonamethecontour..png "panel CSOLabelRenderer to name the contour")
+![CSOLabelRenderer](/images/tutorials/image_processing/CSOLabelRenderer.png " CSOLabelRenderer")
 
+As you can see, the contours are labeled with numbers, so we need to display proper names. To do this, open the `CSOLabelRenderer` panel — it will display the panel shown below.
 
-![Panel View2D after named the contour](/images/tutorials/image_processing/PanelView2Dafternamedthecontour.png "Panel View2D after named the contour")
+ 
+![CSOLabelRenderer panel](/images/tutorials/image_processing/CSOLabelRendererpanel.png "CSOLabelRenderer panel")
 
+Now , add this line to make the numbers names for the contour ?
+labelString = cso.getGroupAt(0).getLabel() 
+then pess apply : 
 
-Now Add SoExaminerViewer and SoCSO3DRenderer as it shown in the below figure to see the 3D image: 
+![Edited CSOLabelRenderer Panel](/images/tutorials/image_processing/contournamed.png " Edited CSOLabelRenderer Panel")
 
-![Output of the SoExaminerViewer](/images/tutorials/image_processing/OutputoftheSoExaminerViewer.png "Output of the SoExaminerViewer")
 
+### **3D Visualization of Contours Using SoExaminerViewer**
 
-**Important information:**
+Now, to view the contours in 3D, follow these steps:
 
+Add the `SoCSO3DRenderer` module and connect it to the `ExtractRTStruc`t module.
+(The `SoCSO3DRenderer` will render the contours (CSOs) in the 3D space.)
 
-* **SoExaminerViewer**
+Add the `SoExaminerViewer` module and connect it to the `SoCSO3DRenderer` module.
+( The `SoExaminerViewer` will allow you to view the 3D contours. You can rotate, zoom, and move around the 3D image.)
+The following figure is the network and the result : 
 
-Shows 3D medical data (`rotate, zoom, and pan`).
+![3D Visualization of Contours Using SoExaminerViewe](/images/tutorials/image_processing/3D.png "3D Visualization of Contours Using SoExaminerViewe")
 
-* **SoCSO3DRenderer**
-Displays CSO contours in 3D view.
 
-Visualizing RTDOSE as a Color Overlay Using LUT
-Add to tour network ( `DicomImportExtraOutput , Soview2DOverlay , SoLUTEditor , Min max scan , Sogroup , Histogram` ) as shown in figure . 
 
+### **Visualizing RTDOSE as a Color Overlay Using LUT**
 
-* **DicomImportExtraOutput**
+Now we need to add another `DicomImportExtraOutput` module to import multiple DICOM objects ,You have to select the right index for the  example it is index 4 thats mean RTDOSE 199*115*147*1 . Add `MinMaxScan`module to scan the input image and updates the minimum and maximum values of the output image  , and connect it with the `DicomImportExtraOutput` 
```

Comment:
extra and missing spaces, comma instead of period - please check spaces and punctuation in all the text

Reply:
In order to show the RTDOSE as an overlay over the original image, we need another...

Always try to explain what you are going to do next, before you describe the steps.